tests/test_iqm_naive_move_pass.py

```diff
@@ -1,85 +1,404 @@
+        validate_circuit(transpiled_circuit, self.backend)
 
-def test_allowed_gates_only(ndonis_architecture):
-    """Test that transpiled circuit has gates that are allowed by the backend"""
-    backend, _client = get_mocked_backend(ndonis_architecture)
-    qubit_registers = _get_qubit_registers(backend)
-    n_qubits = len(qubit_registers)
-    allowed_ops = _get_allowed_ops(backend)
-    for i in range(2, n_qubits + 1):
-        circuit = QuantumVolume(i)
-        transpiled_circuit = transpile_to_IQM(circuit, backend)
-        for instruction in transpiled_circuit.data:
-            assert _is_valid_instruction(transpiled_circuit, allowed_ops, instruction)
+    def test_transpiled_circuit_keeps_layout(self):
+        """Test that the layout of the transpiled circuit is preserved."""
+        layout = [
+            int(qb)
+            for qb in np.random.choice(range(self.backend.num_qubits), self.original_circuit.num_qubits, replace=False)
+        ]
+        transpiled_circuit = self.transpile(initial_layout=layout, optimization_level=3)
+        assert all(x == y for x, y in zip(transpiled_circuit.layout.initial_layout.get_physical_bits(), layout))
+
+
+def test_transpiling_with_unused_qubits(ndonis_architecture):
+    """Test that transpile_to_IQM can handle unused qubits in the circuit."""
 
+    backend = get_mocked_backend(ndonis_architecture)[0]
+    n_qubits = 3
 
-def test_moves_with_zero_state(ndonis_architecture):
-    """Test that move gate is applied only when one qubit is in zero state."""
-    backend, _client = get_mocked_backend(ndonis_architecture)
-    qubit_registers = _get_qubit_registers(backend)
-    n_qubits = len(qubit_registers)
+    # multiple registers, q[1] and x[1:] are unused
+    cr = QuantumRegister(1, "cr")
+    q = QuantumRegister(n_qubits, "q")
+    x = QuantumRegister(3, "x")
+    c = ClassicalRegister(n_qubits, "c")
+    qc = QuantumCircuit(cr, q, x, c)
+
+    qc.rx(np.pi / 2, q[0])
+    qc.append(MoveGate(), [q[0], cr[0]])
     for i in range(2, n_qubits + 1):
-        circuit = QuantumVolume(i)
-        resonator_index = next(
-            j
-            for j, c in enumerate(backend.architecture.components)
-            if c in backend.architecture.computational_resonators
+        if i < n_qubits:
+            qubit = q[i]
+        else:
+            qubit = x[i - n_qubits]
+        qc.rx(np.pi / 2, qubit)
+        qc.cz(qubit, cr[0])
+        qc.rx(-np.pi / 2, qubit)
+    qc.append(MoveGate(), [q[0], cr[0]])
+    qc.barrier()
+    qc.measure(q, c)
+
+    transpile_to_IQM(
+        qc,
+        backend=backend,
+        existing_moves_handling=ExistingMoveHandlingOptions.KEEP,
+    )
+
+
+@pytest.mark.parametrize(
+    "backend,restriction",
+    [
+        ("adonis_architecture", ["QB4", "QB3", "QB1"]),
+        ("ndonis_architecture", ["QB5", "QB3", "QB1", "CR1"]),
+        ("ndonis_architecture", ["QB5", "QB3", "QB1", "CR1"]),
```

Comment:
Duplicated test case.
```suggestion
@pytest.mark.parametrize(
    "backend,restriction",
    [
        ("adonis_architecture", ["QB4", "QB3", "QB1"]),
        ("ndonis_architecture", ["QB5", "QB3", "QB1", "CR1"]),
```